core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -83,15 +82,15 @@ trait RTS {
     ()
   }
 
-  final def schedule[E, A](block: => A, duration: Duration): Async[E, Unit] =
-    if (duration == Duration.Zero) {
+  final def schedule[E, A](block: => A, duration: Duration.Finite): Async[E, Unit] =
```

Comment:
Change to accept infinite durations, in which case, it does not schedule at all (no scheduling!) and simply returns `Async.later`.